app/assets/javascripts/discourse/models/user.js

```diff
@@ -840,10 +840,40 @@ const User = RestModel.extend({
       !secondFactorEnabled &&
       (enforce === "all" || (enforce === "staff" && staff))
     );
+  },
+
+  resolvedTimezone() {
+    if (this._timezone) {
+      return this._timezone;
+    }
+
+    this.set("_timezone", moment.tz.guess());
```

Comment:
If you're not going to use `_timezone` in a computed property or observer (and you shouldn't!) you can omit `set` here and do `this._timezone = `

Reply:
Thanks!